src/main/java/com/epam/javacourses2016/task8/SolverTask8.java

```diff
@@ -10,12 +12,59 @@
     /**
      * Проверяет правильность расстановки скобок.
      * Правильная расстановка:
-     *      1) Каждой открывающей скобке соответствует закрывающая того же типа.
-     *      2) Нет пересечения областей, обрамленных скобками.
+     * 1) Каждой открывающей скобке соответствует закрывающая того же типа.
+     * 2) Нет пересечения областей, обрамленных скобками.
+     *
      * @param string Анализируемая строка.
      * @return true - скобки расставлены верно, иначе - false.
      */
     public boolean isNormalBrackets(String string) {
-        return false;
+        Stack<Character> characters = new Stack<>();
+
+        char[] openTypes = new char[]{'(', '[', '{'};
+        char[] closingTypes = new char[]{')', ']', '}'};
```

Comment:
Ok then
But you are wrong about map is slower. In-array search is linear, search in hash map is constant (if no collisions)